chain/consensus/hierarchical/actors/subnet/subnet_actor.go

```diff
@@ -143,6 +145,18 @@ func (a SubnetActor) Join(rt runtime.Runtime, _ *abi.EmptyValue) *abi.EmptyValue
 		}
 	}
 
+	rt.StateTransaction(&st, func() {
+		// Mutate state
+		if st.Consensus == hierarchical.Mir {
```

Comment:
Why not making this functionality general and not Mir-specific? We can set `MinValidators` to be 0 by default, but having this for every consensus algorithm would enable interesting use cases (like for instance requiring a minimum number of PoW miners before starting a network to prevent the first joiner from having an unfair advantage).

Reply:
fixed